src/Utils/obiUtils.tsx

```diff
@@ -46,7 +46,7 @@ export async function getLogDialogActivities(
     conversationId: string | undefined,
     channelId: string | undefined,
     fetchLogDialogThunkAsync: (appId: string, logDialogId: string, replaceLocal: boolean, nullOnNotFound: boolean, noSpinner: boolean) => Promise<CLM.LogDialog>,
-    fetchActivitiesAsync: (appId: string, trainDialog: CLM.TrainDialog, userName: string, userId: string, useMarkdown: boolean) => Promise<CLM.TeachWithActivities>
+    fetchActivitiesAsync: (appId: string, trainDialog: CLM.TrainDialog, userName: string, userId: string, useMarkdown: boolean, noSpinner: boolean) => Promise<CLM.TeachWithActivities>
```

Comment:
These could really use some documentation.  Please flesh out the function comments and add parameter docu.

Reply:
We've generally not been documenting parameters when they are self-explanatory, but they may just seem more self-explanatory to us as we've been working with the code base for a long time

Reply:
I don't think documentation fixes root of issue here. The functions like this should take objects to have `named`  parameters. The `, true, true, true)` is very bad as consumer doesn't know what each `true` means. Would be much better as

```
fetchLogDialogThunkAsync({
  appId,
  logDialogId,
  replaceLocal: true,
  nullOnNotFound: true,
  noSpinner: true
})
```

This is one of the reasons they added short-hand parameters to objects.

Reply:
In this case, I think the thunk has enough params that it could use documentation.  I have no idea from looking at the signature why `userName` is required, or who it refers to; or why the `noSpinner` param is relevant for example.

Reply:
Right place to document this would be in logActions.   All our our actions have this issue....

Reply:
I'm all for self-documenting code, but can also attest that as somebody new to the project, lack of actual documentation has been a **large** source of friction for getting spun up.

If you go with Matt's idea of using an object to store the params, that would make a convenient place to add documentation.

Reply:
Added comments to thunks

Reply:
If we pass objects to the thunks we should change all of them at the same time.  It's weird to have one outlier

Reply:
Wow! Going from "I change things gradually" and intentionally introduce outliers to "we should change all of them" "It's weird to have one outlier". Great change, but I will believe it when I see it.

Reply:
Grazie!  For my own knowledge, though, who/what is the user referenced in the thunk params ?  I haven't yet seen us passing around users elsewhere in the code.

Reply:
The UI uses a simulated `user` object `{ name, id }` to recreate what happens when user is actually logged into something like Teams, Skype, etc.  All those channels would send this user object on the activity. Since we know this activity since it deterministic from the SDK the SDK can then check for it and know if it activity is coming from the administrative UI or some actual channel and treat it differently.